MNT towards removing assert_equal, etc

adrinjalali opened the pull request:
Towards #14215 

This moves the codebase towards removing the assert_equal, etc.

Some instances are still left, since the nose2pytest did not fix them.

adrinjalali committed:
fix cluster and compose

adrinjalali committed:
fix common tests

adrinjalali committed:
fix covariance, cross_decomposition, datasets

adrinjalali committed:
fix decomposition

adrinjalali committed:
fix ensemble

adrinjalali committed:
fix feature_extraction

adrinjalali committed:
fix feature_selection

adrinjalali committed:
fix gaussian_processes

adrinjalali committed:
fix linear_model

adrinjalali committed:
fix manifold

adrinjalali committed:
fix metrics

adrinjalali committed:
cleanup assert_message

adrinjalali committed:
fix mixture, model_selection

adrinjalali committed:
fix neighbors

adrinjalali committed:
fix neural_networks

adrinjalali committed:
fix preprocessing

adrinjalali committed:
fix semi_supervised, svm

adrinjalali committed:
fix semi_supervised, tree

adrinjalali committed:
fix utils

adrinjalali committed:
pep8

adrinjalali committed:
remove some unused imports

adrinjalali committed:
automatic removal of unused imports

jnothman reviewed (approved):
These are all automatic, yeah?

adrinjalali commented:
`nose2pytest` doesn't properly handle the case when you have:

```python
assert_equal(condition, "starting "
                        "a very long message")
```

It generates invalid code, and those are the ones I manually fixed, and then there were a bunch of pep8 issues that I fixed separately mostly in https://github.com/scikit-learn/scikit-learn/pull/14222/commits/0e5d4e787a796fcf89e3112511fb95d45bb7d6bf

Then there was removal of the unused imports, which `autoflake` does, but it also removes the unnecessary `pass` lines. Since I didn't want to touch those ones in this PR, I modified `autoflake` to ignore those `pass`s and they're kept in the code.
